wled00/json.cpp

```diff
@@ -809,23 +809,31 @@ void serveJson(AsyncWebServerRequest* request)
     return;
   }
 
+#ifdef WLED_USE_DYNAMIC_JSON
   AsyncJsonResponse* response = new AsyncJsonResponse(JSON_BUFFER_SIZE);
-  JsonObject doc = response->getRoot();
+#else
+  while (jsonBufferLock) delay(1);
```

Comment:
The more I think about it, the more I think this can go horribly wrong in case JSON is requested via network while `jsonBufferLock` is already true from a loop task because the network interrupt happened just while another thing was using the buffer. As far as I know `delay()` does not work correctly in the system/interrupt context, and even if it did, it would lock up as it would not return to the main loop to finish the task.

Of course it seems to work just fine regardless (perhaps because the time the doc is locked is short), but somehow I don't have a good feeling about it.

I have two ideas how we could guarantee for everything to not intersect:
1) use `noInterrupts()` and `interrupts()` before and after every use of doc in the loop. This could have some very hard to debug side effects though if something within that code relies on interrupts, or interrupts can't get executed in time
2) Instead of locking (one loop task using doc can't intersect another anyways, as they are all run sequentially), set a dirty flag from the interrupt to let the loop know that it needs to clean the document and start over. If e.g. a Serial JSON command gets dropped because of that, that is way less critical than a potential crash. The only thing I'd be worried about is preset saving, would be bad for the doc to go while it is being written to the FS...

https://arduino.stackexchange.com/questions/70369/disable-interrupts-when-doing-critical-things is interesting

Reply:
Indeed. I have noticed 2 cases of lockups (with no apparent requests happening but I do have MQTT enabled).
This will need serial debugging which I will do in the next few days.

Reply:
One thing that crosses my mind is to add a time-out option to the loop as to not lock up ESP.
Next would be try/catch handling if buffer is still locked after timeout (same behaviour as out of memory condition) with more graceful outcome.

Reply:
A few bugs found and squashed. No more lockups.